blueflood-core/src/main/java/com/rackspacecloud/blueflood/utils/QueryDiscoveryModuleLoader.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2015 Rackspace
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+package com.rackspacecloud.blueflood.utils;
+
+import com.rackspacecloud.blueflood.io.DiscoveryIO;
+import com.rackspacecloud.blueflood.service.Configuration;
+import com.rackspacecloud.blueflood.service.CoreConfig;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+
+public class QueryDiscoveryModuleLoader {
```

Comment:
@dlobue It performs some checks which are useful only if you are using it for the `Query Module` like checking and throwing an exception if someone specifies multiple discovery modules while starting query service, but for ingestion modules it makes sense to have multiple discovery modules and inserting metrics for each one of them.